gpudrive/env/env_puffer.py

```diff
@@ -422,19 +422,32 @@ def render_env(self):
             np.where(np.array(list(self.rendering_in_progress.values())))[0]
         )
         time_steps = list(self.episode_lengths[envs_to_render, 0])
-
+        
         if len(envs_to_render) > 0:
-            sim_state_figures = self.env.vis.plot_simulator_state(
-                env_indices=envs_to_render,
-                time_steps=time_steps,
-                zoom_radius=self.zoom_radius,
-            )
+            if self.render_backend == "matplotlib":
+                # Render bird's eye view using matplotlib
+                sim_state_figures = self.env.vis.plot_simulator_state(
+                    env_indices=envs_to_render,
+                    time_steps=time_steps,
+                    zoom_radius=self.zoom_radius,
+                )
 
-            for idx, render_env_idx in enumerate(envs_to_render):
+                for idx, render_env_idx in enumerate(envs_to_render):
+                    self.frames[render_env_idx].append(
+                        img_from_fig(sim_state_figures[idx])
+                    )
+            else:
+                # Render bird's eye view using raster scan algorithm 
+                bev = self.env.sim.bev_observation_tensor().to_torch()
+                # Convert the BEV observation to a colored segmentation map
+                # If the tensor is one-hot encoded segmentation data
+                colored_bev = color_onehot_segmentation_map(bev, 'cpu')
+                agent_idx = 0
```

Comment:
This is always the `sdc_index` right? (I think so if I am following `json_serialization.hpp` well. 
Maybe you would call this variable `sdc_index` just to it is clear we are logging that agent?

Reply:
yes, the 0th index is always the sdc 

Note: The implementation doesn't work yet: the raster image I get is black and white. Also, do you think it would be possible to display the raster image in the case when we control multiple agents?

Reply:
I think you need to permute the bev map so that the number of classes is the first dimension
`bev.permute(2, 0, 1)` or something like this.

For your second question, if im understanding correctly, I dont see an issue, you just might have to modify the colorization function to accept a first batch dimesion.